### PR TITLE
feat: user is able to show/hide card and apply the change to all user…

### PR DESCRIPTION
**Pull request by alexunique0519**
…s in the same room

##[DUMS-75](https://jira-dev.bdm-dev.dts-stn.com/browse/DUMS-75)
##[DUMS-77](https://jira-dev.bdm-dev.dts-stn.com/browse/DUMS-77)

### Description
On the client side, user could click to show/hide button to call mutations to change the current card' show/hide status. 

List of proposed changes:

-
-

### CODE REVIEW

File: pages/room/[id].js [hunk 1]
```diff
@@ -75,6 +75,13 @@ export default function Room(props) {
     variables: { room: props.roomId },
   })
 
+  const roomSubscription = useSubscription(ROOM_SUBSCRIPTION, {
+    variables: { room: props.roomId },
+  })
+
+  //const [showHideCard] = useMutation(SHOW_HIDE_ROOM_CARD)
+  const [showHideCard] = useMutation(UPDATE_ROOM)
```
Comment (krischarbonneau) [inline pages/room/[id].js:83]:
```suggestion
  const [updateRoom] = useMutation(UPDATE_ROOM)
```

This should be more generic as we'll likely be using this mutation for other features

**Review by krischarbonneau — CHANGES_REQUESTED** [review]

File: pages/room/[id].js [hunk 2]
```diff
@@ -98,7 +105,18 @@ export default function Room(props) {
...
   }, [userSubscription])
 
+  useEffect(() => {
+    if (roomSubscription.data) {
+      setHidden(!roomSubscription.data.roomUpdated.isShown)
+    }
+  }, [roomSubscription])
+
   if (!pageState && users) {
+    const usersIDList = []
```
Comment (DaceyTom2) [inline pages/room/[id].js:115]: Rework this to use the room data. users and room might become inconsistent. Also this constantly adds the current user to the end of the users list ie [ 1, 2, 3, 1, 1, 1, 1]

**Review by DaceyTom2 — CHANGES_REQUESTED** [review]
Couple of things. Let me know if you want to set up a call to go over them as they are hard to explain with typing

**Comment by DaceyTom2** [comment]
Big improvement!

**Review by DaceyTom2 — APPROVED** [review]

**Review by krischarbonneau — APPROVED** [review]

**Pull request merged**